### PR TITLE
Add scalac warnings for unused imports

### PR DESCRIPTION
**Pull request by jw3**
- cleans up unused imports using scalafix
- enables scalac warn-on-unused option
- adds support for version specific scalac options
- enables language existentials in 2.11
- does not organize imports, only removes unused

DAFFODIL-2145

### CODE REVIEW

**Comment by jw3** [comment]
Ah, yeah will need separate options for 2.11...

File: daffodil-runtime1/src/main/scala/org/apache/daffodil/udf/UserDefinedFunctionService.scala [hunk 1]
```diff
@@ -28,7 +28,6 @@ import java.lang.reflect.Method
 import org.apache.daffodil.dpath.NodeInfo
 import java.io.Serializable
 import java.io.ObjectInputStream
-import scala.language.existentials
```
Comment (jw3) [inline daffodil-runtime1/src/main/scala/org/apache/daffodil/udf/UserDefinedFunctionService.scala:31]: Had to add a language feature flag for 2.11 due to the import being seen as unused in 2.12
  - See https://github.com/scala/bug/issues/11467

**Comment by tuxji** [comment]
Looks good to me.

I followed the [scala/bug#11467](https://github.com/scala/bug/issues/11467) link, wondered who sjrd was, googled, found out he authored Scala.js, browsed his Twitter page, and learned about some interesting stuff including [scalafix-organize-imports](https://github.com/liancheng/scalafix-organize-imports).  Seems even more powerful than scalafix's RemoveUnused rule since it organizes all imports nicely into groups, although only one or the other rule should be used.

**Comment by jw3** [comment]
It needs squashed some.  Do you want two commits with the build.sbt changes left separate, or squash it all into one?

**Comment by stevedlawrence** [comment]
I think one commit is fine in this case. It's maybe a little difficult for the important build.sbt change to stand out, but it is usually going to be the first file listed in a diff so it's not too hidden.

**Comment by tuxji** [comment]
+1  Steve encouraged me to try out my committer bit :-).

Once your squashed commit passes the checks, I'll see if I can merge your commit too.

**Pull request merged**